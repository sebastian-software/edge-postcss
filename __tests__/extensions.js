import { compile } from "./core"

// ====================================================
// ================= EXTENSIONS =======================
// ====================================================

test("Responsive Type", () =>
  compile("html { font-size: responsive 12px 21px; font-range: 420px 1280px; }")
)

test("Clearfix", () =>
  compile(".row { clear: fix; }")
)

test("System UI", () =>
  compile("body { font-family: system-ui; }")
)

test("Normalize.css", () =>
  compile(".before { color: red; } @import-normalize; .after { color: green; }")
)

test("Initial", () =>
  compile("h1 { font-family: initial; }")
)

test("Initial - All", () =>
  compile("ul { all: initial; }")
)
