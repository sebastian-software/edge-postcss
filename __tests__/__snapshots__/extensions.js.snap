// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clearfix 1`] = `
".row:after{
content:'';
display:block;
clear:both
}"
`;

exports[`Initial - All 1`] = `
"ul{
border-collapse:separate;
border-spacing:0;
caption-side:top;
cursor:auto;
direction:ltr;
empty-cells:show;
font-family:serif;
font-size:medium;
font-style:normal;
font-variant:normal;
font-weight:400;
font-stretch:normal;
line-height:normal;
-webkit-hyphens:none;
-ms-hyphens:none;
hyphens:none;
letter-spacing:normal;
list-style:disc outside none;
-moz-tab-size:8;
tab-size:8;
text-align:left;
-moz-text-align-last:auto;
text-align-last:auto;
text-indent:0;
text-shadow:none;
text-transform:none;
visibility:visible;
white-space:normal;
widows:2;
word-spacing:normal;
all:initial
}"
`;

exports[`Initial 1`] = `
"h1{
font-family:initial
}"
`;

exports[`Normalize.css 1`] = `
".before{
color:#ff4136
}

html{
line-height:1.15;
-ms-text-size-adjust:100%;
-webkit-text-size-adjust:100%
}

h1{
font-size:2em;
margin:.67em 0
}

hr{
box-sizing:content-box;
height:0;
overflow:visible
}

details,main{
display:block
}

code,kbd,pre,samp{
font-family:monospace,monospace;
font-size:1em
}

a{
background-color:transparent;
-webkit-text-decoration-skip:objects
}

abbr[title]{
-webkit-text-decoration:underline dotted;
text-decoration:underline dotted
}

b,strong{
font-weight:bolder
}

small{
font-size:80%
}

img{
border-style:none
}

svg:not(:root){
overflow:hidden
}

button,input,optgroup,select,textarea{
margin:0
}

button{
overflow:visible;
text-transform:none
}

[type=button],[type=reset],[type=submit],button{
-webkit-appearance:button
}

[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{
border-style:none;
padding:0
}

[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{
outline:1px dotted ButtonText
}

fieldset{
padding:.35em .75em .625em
}

input{
overflow:visible
}

legend{
color:inherit;
display:table;
max-width:100%;
white-space:normal
}

progress{
display:inline-block;
vertical-align:baseline
}

select{
text-transform:none
}

textarea{
overflow:auto
}

[type=checkbox],[type=radio],legend{
box-sizing:border-box;
padding:0
}

[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{
height:auto
}

[type=search]{
-webkit-appearance:textfield;
outline-offset:-2px
}

[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{
-webkit-appearance:none
}

::-webkit-file-upload-button{
-webkit-appearance:button;
font:inherit
}

summary{
display:list-item
}

[hidden],template{
display:none
}

.after{
color:#2ecc40
}"
`;

exports[`Responsive Type 1`] = `
"html{
font-size:calc(7.60465px + 1.04651vw)
}

@media screen and (min-width:1280px){
html{
font-size:21px
}
}

@media screen and (max-width:420px){
html{
font-size:12px
}
}"
`;

exports[`System UI 1`] = `
"body{
font-family:system-ui,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue
}"
`;
